examples/knowledge_tuning/knowledge_utils.py

```
@@ -97,10 +97,56 @@ def _conv_pretrain(rec):
     ]
     return rec
 
+def mask_qa_per_doc(ds, keep_no_qa_per_doc=3):
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)